engine/src/execution_graph/logic_controllers/BatchProcessing.cpp

```diff
@@ -266,6 +272,11 @@ kstatus BindableTableScan::run() {
         std::iota(projections.begin(), projections.end(), 0);
     }
 
+    if (provider->is_sql()) {
+      auto sql_provider = std::dynamic_pointer_cast<ral::io::abstractsql_data_provider>(provider);
```

Comment:
same as comment right above, this should be set at construction

Reply:
done